### PR TITLE
refactor(AdminTags): refactor code in admin tags

### PR DESCRIPTION
**Pull request by cerkiewny**
use functional components for modal window and move to final forms in tag selection

PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Moved the admin tag UI to the react-final-form, also refactored it heavily and introduced better error handling

## Screenshots/Videos

image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -109,7 +109,7 @@
     "react": "17.0.2",
     "react-dom": "17.0.2",
     "react-dropzone": "^10.1.10",
-    "react-final-form": "6.5.3",
```
Comment (thisislawatts) [inline package.json:112]: **question:** Is this change required?

Comment (cerkiewny) [reply package.json:112]: probably not :P

Comment (thisislawatts) [reply package.json:112]: Okay thanks for confirming, please can you remove this from the PR.

**Comment by thisislawatts** [comment]
Thanks for the update @cerkiewny, would you be able to rebase to resolve the merge conflict?

**Comment by cerkiewny** [comment]
sure, will do

**Comment by AlfonsoGhislieri** [comment]
Is this good for you to go @thisislawatts ?

**Review by thisislawatts — APPROVED** [review]

**Comment by thisislawatts** [comment]
Thanks for contributing ✨ Please @all-contributors add @cerkiewny for code
